honeybee_radiance_folder/folderutil.py

```diff
@@ -29,33 +29,38 @@ def from_dict(cls, input_dict):
         .. code-block:: python
 
             {
-            "name": "grass_covered",
+            "identifier": "grass_covered",
             "default": "ground..summer..000.rad",
             "direct": "ground..direct..000.rad",
             }
         """
-        # TODO: add checks for keys
-        name = input_dict['name'] if 'name' in input_dict else None
-        default = input_dict['default']
-        direct = input_dict['direct']
-        return cls(name, default, direct)
+        for key in ['identifier', 'default', 'direct']:
+            assert key in input_dict, 'State is missing required key: %s' % key
+        identifier = input_dict['identifier']
+        default = os.path.normpath(input_dict['default'])
+        direct = os.path.normpath(input_dict['direct'])
+        return cls(identifier, default, direct)
 
     def validate(self, folder):
-        """Validate files in this state."""
+        """Validate files in this state.
+
+        Args:
+            folder: Path to state folder.
+        """
         assert os.path.isfile(os.path.join(folder, self.default)), \
-            'Failed to find default file for %s' % self.name
+            'Failed to find default file for %s' % self.identifier
         assert os.path.isfile(os.path.join(folder, self.direct)), \
-            'Failed to find direct file for %s' % self.name
+            'Failed to find direct file for %s' % self.identifier
 
     def __repr__(self):
-        return 'NonApertureState: {}'.format(self.name)
+        return 'NonApertureState: {}'.format(self.identifier)
 
 
 class ApertureState(NonApertureState):
     """A state for a dynamic aperture from Radiance files.
 
     Args:
-        name (str): Optional human-readable name for the state. Can be None.
+        identifier (str): Optional human-readable identifier for the state. Can be None.
```

Comment:
It might be good to say in the docstring that this object parallels the honeybee-radiance `SubFaceState` just to make it clear to people familiar with honeybee-radiance that these classes refer to the same thing.

Reply:
Or just let me know if you think honeybee-radiance should be changed. I called it `SubFaceState` because it can be assigned to both `Apertures` and `Doors`.

Reply:
I think you picked the right name for inside honeybee-library but for the folder I think ApertureState is more appropriate. I will add the note as you suggested.